.github/scripts/open-prod-pr.mjs

```diff
@@ -0,0 +1,27 @@
+import { exec } from "node:child_process";
+import { promisify } from "node:util";
+
+const promiseExec = promisify(exec);
+
+const branchesDiffResult = await promiseExec("git log main..canary").catch((e) => {
+  console.error(e);
+  process.exit(1);
+});
+
+if (branchesDiffResult.stdout === "") {
+  console.log("Branches canary and main have no different commits");
+  process.exit(0);
+} else if (branchesDiffResult.stdout.length > 0) {
+  console.log("Found different commits, open a PR");
+  await promiseExec(
+    "gh pr create -B main -H canary --title 'Merge canary to main' --body 'Merge canary to main, to trigger a prod release"
+  )
```

Comment:
That looks very dangerous to me if someone decides to add arbitrary arguments in the future. I highly recommend using `child_process.spawn` (+ stderr & stdout pipes for logging purposes).

```js
child_process.spawn(
  "gh",
  [
    "pr", "create", 
    "-B", "main", 
    "-H", "canary", 
    "--title", "Merge canary to main", 
    "--body", "Merge canary to main, to trigger a prod release"
  ]
)
```

Reply:
rewritten to spawn